astropy/io/ascii/sextractor.py

```diff
@@ -64,6 +67,15 @@ def get_cols(self, lines):
                     column_unit = columns[previous_column][2]
                     columns[c] = (column_name, column_descr, column_unit)
             previous_column = n
+        if dataline is not None and len(colnumbers) > 0:  # handle the case where the last column is array-like
```

Comment:
I was wondering about a slightly different strategy to avoid repeating code.  After the point of defining `colnumbers`, what about something like:
```
# If there are more data columns than the largest column number
# then add a pseudo-column that will be dropped later.  This allows
# the array column logic below to work in all cases.
n_data_cols = len(dataline.split())
if n_data_cols > colnumbers[-1]:
    columns[n_data_cols] = (None, None, None)
    colnumbers.append(n_data_cols)
```
This adds a pseudo-column at the end so that the normal array processing below works.  Then toward the end you can just ignore any column with `columns[n][0] is None`.

I haven't thought it through it in complete detail but this might be slightly cleaner without the code repetition.